addon/store.js

```diff
@@ -46,45 +46,39 @@ var Store = Ember.Object.extend({
       this.set("array", {});
     },
     clear(type) {
-        if(type === undefined) {
+        if (type === undefined) {
             this.reset();
         }
+
         delete this.get("identityMap")[type];
         arrayForType(type, this).clear();
-
-        run.once(() => {
```

Comment:
@jasonmit for reference - the perf app I used below to benchmark this (the original app came from @matthewnessworthy btw)

https://github.com/toranb/ember-cli-simple-store-perf

These are the times I see this morning w/ 4K in the store

![screen shot 2016-01-18 at 8 12 45 am](https://cloud.githubusercontent.com/assets/147411/12393471/7354dcfa-bdbb-11e5-8937-4dd203238c14.png)